### DISCUSSION TITLE
How to set a node as read only ?

### DISCUSSION THREAD

**Discussion by iceweb99** [freeplane/freeplane#830] [Help]
Is it possible to set a node as read only ? I want to avoid changing the node content by accident. If not is there a workaround to accomplish the same ?

Regards.

**Comment by euu2021** [comment]
I don't think so.

Possible workarounds:

- set a password to the node and lock it. The shortcoming is that you won't be able to see the descendants
- create a property "read only"; create a Node Listener that, after each node change, checks is the property is present; is yes, it does something like: gives a warning, or even reverts the change automatically

**Reply by dpolivaev** [reply]
I am afraid that password protection is only applied to the descendants but not to the node itself.
Currently, Freeplane supports read-only mode for whole mind maps but not for single nodes.
Theoretically, such support could be implemented but I am afraid too few users need it and even much less users would find it.

**Reply by euu2021** [reply]
> I am afraid that password protection is only applied to the descendants but not to the node itself.

My bad. I tried creating a node inside a locked node and got the message "Target node is write protected", so I deduced that it would also be impossible to change the node core.

**Reply by iceweb99** [reply]
Thanks euu2021 for your response. I feel your second option would be something to look at once I get more familiar with Freeplane.

> 
>    create a property "read only"; create a Node Listener that, after each node change, checks is the property is present; is yes, it does something like: gives a warning, or even reverts the change automatically

**Reply by iceweb99** [reply]
Thanks dpolivaev, I hope that more users vote for it. The password protection icon did feel at first like it was the read-only mode option for the node, took me some time to understand that it works differently. 

> I am afraid that password protection is only applied to the descendants but not to the node itself. Currently, Freeplane supports read-only mode for whole mind maps but not for single nodes. Theoretically, such support could be implemented but I am afraid too few users need it and even much less users would find it.

**Comment by iceweb99** [comment]
I have been actively using Freeplane for the last few days and these are my learnings.

There were two things due to which I felt that the content in the nodes could be easily lost/overwritten.
1. Node selection as soon as the mouse pointer moves over it.
2. Node content gets overwritten in the selected node as soon as any key is pressed.

Changing the behaviour of the above two settings made it easier for me to feel that the data is not so volatile.

The settings are at 
1. Tools menu > Preferences > Behaviour > (In "Selection method" section) Selection method = By click
2. Tools menu > Preferences > Behaviour > (In "Behaviour" section) On key type = nothing

**Reply by josephs07** [reply]
you can avoid accidental deletion or subtrees being cut by unchecking these settings at (menu > Preferences >Confirmations)
<img width="483" height="187" alt="image" src="https://github.com/user-attachments/assets/e7a93175-a940-47c7-b345-c917d97765af" />

**Comment by maximebochon** [comment]
I have created a related feature request : #2725 [Add a user function to freeze/unfreeze changes on a node ](https://github.com/freeplane/freeplane/issues/2725)
